### PR TITLE
[MM-16751] golint model

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This PR enabled the `golint` check for `model`.

The main migration task was to rename the constants from `SOME_CONST_NAME` to `SomeConstName`.

This was accomplished by a small go script which uses  `go/ast`. You can find the code [here](https://gist.github.com/hanzei/d255a3797bcb0f1dbfa7a0770de7d3af). Use `for f in $(find . -iname "*.go"); do ast-golint -f $f; goimports -w $f; done` to run it recursively from the current directly. (Props to https://xdg.me/rewriting-go-with-ast-transformation/ for a great introduction into rewriting go code)

The programmatic migration took place in https://github.com/mattermost/mattermost-server/commit/48295c00bd456bd570109aa39d7fd8da8342e368 and https://github.com/mattermost/mattermost-server/commit/093dda69b11ed7c12ea8966208c0a35707f667e3.

https://github.com/mattermost/mattermost-server/commit/c550d43ee7c9203c28ced251f6f5389f3c28bfaa contains manual fixes in `model` for corner cases like renaming `NoticeInstanceType_Both` to `NoticeInstanceTypeBoth` and fixing the comments.

https://github.com/mattermost/mattermost-server/commit/563b5c4324bfaa0fb3f78baaefc0e6a8d5fa658c contain the manual fixes for corner cases in the rest of the code base.

Due to a naming conflict with the `PostEphemeral` `struct` I've renamed the post types to include the word `Type` e.g. `PostAddToTeam` is now  `PostTypeAddToTeam`. This was done using `gorename` and took place in https://github.com/mattermost/mattermost-server/commit/8461b491a220927822dc4648b084e22977ba862c alongside some naming fixes for `Oauth` -> `OAuth` and `Sso` -> `SSO`.

I applied similar naming changes for the channel type in .https://github.com/mattermost/mattermost-server/commit/8461b491a220927822dc4648b084e22977ba862c `PostSlackAttachment` is now `PostTypeSlackAttachment`.

Finally, https://github.com/mattermost/mattermost-server/commit/e0a3953390832489cb58389609397709605d4271 contains a fix for a plugin test that depends on `model`.

Please note that I didn't check _if_ the constants need to be exported. I also didn't fix all the comments in the code base as the chances of a false positive change seemed to high to me.

Further capitalization fixes will happen as part of https://mattermost.atlassian.net/browse/MM-23260, e.g. `Ip` -> `ÌP`.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-16751

#### Release Note
```release-note
Apply golint to model package
```

### CODE REVIEW

**Comment by hanzei** [comment]
@mkraft @agnivade I wasn't quite sure who the best persons to review are. Let me know if you don't feel suitable to review these changes.

**Comment by hanzei** [comment]
I accidentally pushed unrelated changes to this branch. Fixed via a force push.

File: model/post.go [hunk 1]
```diff
@@ -18,56 +18,58 @@ import (
...
+	PostPropsDeleteBy          = "deleteBy"
+	PostPropsOverrideIconUrl   = "override_icon_url"
+	PostPropsOverrideIconEmoji = "override_icon_emoji"
+
+	PostPropsMentionHighlightDisabled = "mentionHighlightDisabled"
+	PostPropsGroupHighlightDisabled   = "disable_group_highlight"
 )
 
-var AT_MENTION_PATTEN = regexp.MustCompile(`\B@`)
+var AtMentionPatten = regexp.MustCompile(`\B@`)
```
Comment (agnivade) [inline model/post.go:72]: Finally, we can correct the spelling mistake here :) Let's change it to `AtMentionPattern`.

File: model/user.go [hunk 2]
```diff
@@ -23,41 +23,41 @@ import (
...
-	USER_POSITION_MAX_RUNES   = 128
-	USER_FIRST_NAME_MAX_RUNES = 64
-	USER_LAST_NAME_MAX_RUNES  = 64
-	USER_AUTH_DATA_MAX_LENGTH = 128
-	USER_NAME_MAX_LENGTH      = 64
-	USER_NAME_MIN_LENGTH      = 1
-	USER_PASSWORD_MAX_LENGTH  = 72
-	USER_LOCALE_MAX_LENGTH    = 5
-	USER_TIMEZONE_MAX_RUNES   = 256
+	ME                             = "me"
```
Comment (agnivade) [inline model/user.go:26]: Same as earlier - should this be "Me"?

**Review by agnivade — COMMENTED** [review]
Great work! Just some small comments. :)

File: api4/ldap.go [hunk 3]
```diff
@@ -87,8 +87,8 @@ func testLdap(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func getLdapGroups(c *Context, w http.ResponseWriter, r *http.Request) {
-	if !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PERMISSION_SYSCONSOLE_READ_USERMANAGEMENT_GROUPS) {
-		c.SetPermissionError(model.PERMISSION_SYSCONSOLE_READ_USERMANAGEMENT_GROUPS)
+	if !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionSysconsoleReadUsermanagementGroups) {
+		c.SetPermissionError(model.PermissionSysconsoleReadUsermanagementGroups)
```
[the remark this answers is not in the document]
Comment (hanzei) [reply api4/ldap.go:91]: Fixed in e45c1a050.

File: api4/license.go [hunk 4]
```diff
@@ -218,7 +218,7 @@ func requestTrialLicense(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	trialLicenseRequest := &model.TrialLicenseRequest{
 		ServerID:              c.App.TelemetryId(),
-		Name:                  currentUser.GetDisplayName(model.SHOW_FULLNAME),
+		Name:                  currentUser.GetDisplayName(model.ShowFullname),
```
[the remark this answers is not in the document]
Comment (hanzei) [reply api4/license.go:221]: Fixed the bulk of it in b466761de.

There are some uses of `Fullname` in packages that deal with external services like bleve and ES. Can we just rename them? https://github.com/mattermost/mattermost-server/blob/2b6c0e974690414e4b2df86962667df65af3abd0/services/searchengine/bleveengine/common.go#L19-L25

File: api4/oauth_test.go [hunk 5]
```diff
@@ -535,7 +535,7 @@ func TestGetAuthorizedOAuthAppsForUser(t *testing.T) {
 	CheckNoError(t, resp)
 
 	authRequest := &model.AuthorizeRequest{
-		ResponseType: model.AUTHCODE_RESPONSE_TYPE,
+		ResponseType: model.AuthcodeResponseType,
```
[the remark this answers is not in the document]
Comment (hanzei) [reply api4/oauth_test.go:538]: Fixed in ddc4a9af0.

File: api4/license.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (agnivade) [reply api4/license.go:221]: Yeah, I think those can be HW tickets. It's slightly out of scope for the v6 changes.

File: model/client4.go [hunk 6]
```diff
@@ -19,35 +19,35 @@ import (
...
+	HeaderCsrfToken          = "X-CSRF-Token"
+	HeaderBearer             = "BEARER"
+	HeaderAuth               = "Authorization"
+	HeaderCloudToken         = "X-Cloud-Token"
+	HeaderRemoteclusterToken = "X-RemoteCluster-Token"
+	HeaderRemoteclusterId    = "X-RemoteCluster-Id"
+	HeaderRequestedWith      = "X-Requested-With"
+	HeaderRequestedWithXml   = "XMLHttpRequest"
+	HeaderRange              = "Range"
+	STATUS                   = "status"
```
[the remark this answers is not in the document]
Comment (hanzei) [reply model/client4.go:40]: The name `Status` is already occupied from https://github.com/mattermost/mattermost-server/blob/48295c00bd456bd570109aa39d7fd8da8342e368/model/status.go#L22-L30.

I would like to rename it to `HeaderStatus` which makes sense for most uses, but it's also used as a status body in e.g. https://github.com/mattermost/mattermost-server/blob/093dda69b11ed7c12ea8966208c0a35707f667e3/web/web.go#L103-L107. Hence, I'm incident to deffer the fix to a different PR as some "real" changes would be needed here in the form of decoupling the status body and the header.

File: model/cluster_discovery.go [hunk 7]
```diff
@@ -11,8 +11,8 @@ import (
 )
 
 const (
-	CDS_OFFLINE_AFTER_MILLIS = 1000 * 60 * 30 // 30 minutes
-	CDS_TYPE_APP             = "mattermost_app"
+	CdsOfflineAfterMillis = 1000 * 60 * 30 // 30 minutes
+	CdsTypeApp            = "mattermost_app"
```
[the remark this answers is not in the document]
Comment (hanzei) [reply model/cluster_discovery.go:15]: Fixed in eb5460efa.

File: model/post.go (hunk 1 above)
Comment (hanzei) [reply model/post.go:72]: Fixed in c9f571566

File: model/user.go (hunk 2 above)
Comment (hanzei) [reply model/user.go:26]: Fixed in 391eefe85.

**Comment by hanzei** [comment]
Thanks for the review @agnivade.  :+1: PTAL

**Review by agnivade — APPROVED** [review]
Brilliant work Ben!

**Review by streamer45 — APPROVED** [review]
This is a great effort @hanzei :tada: Also love the use of automation for this :+1:  
Agree we can take care of initialisms and other minor modifications in a separate PR.

Just left some non-blocking comments/suggestions.

File: model/file_info.go [hunk 8]
```diff
@@ -15,8 +15,8 @@ import (
 )
 
 const (
-	FILEINFO_SORT_BY_CREATED = "CreateAt"
-	FILEINFO_SORT_BY_SIZE    = "Size"
+	FileinfoSortByCreated = "CreateAt"
+	FileinfoSortBySize    = "Size"
```
[the remark this answers is not in the document]
Comment (hanzei) [reply model/file_info.go:19]: I don't think that possible without breaking comparability with existing plugins as there is a  `GetFileInfos` API method.

File: model/license.go [hunk 9]
```diff
@@ -12,16 +12,16 @@ import (
...
+	LicenseGracePeriod  = 1000 * 60 * 60 * 24 * 10 //10 days
+	LicenseRenewalLink  = "https://mattermost.com/renew/"
 )
 
 const (
-	SIXTY_DAYS                        = 60
-	FIFTY_EIGHT                       = 58
-	LICENSE_UP_FOR_RENEWAL_EMAIL_SENT = "LicenseUpForRenewalEmailSent"
+	SixtyDays                    = 60
+	FiftyEight                   = 58
```
[the remark this answers is not in the document]
Comment (hanzei) [reply model/license.go:23]: Removed them in 92d112f38.

File: model/link_metadata.go [hunk 10]
```diff
@@ -16,10 +16,10 @@ import (
...
 
 const (
-	LINK_METADATA_TYPE_IMAGE     LinkMetadataType = "image"
-	LINK_METADATA_TYPE_NONE      LinkMetadataType = "none"
-	LINK_METADATA_TYPE_OPENGRAPH LinkMetadataType = "opengraph"
-	MAX_IMAGES                   int              = 5
+	LinkMetadataTypeImage     LinkMetadataType = "image"
+	LinkMetadataTypeNone      LinkMetadataType = "none"
+	LinkMetadataTypeOpengraph LinkMetadataType = "opengraph"
+	MaxImages                 int              = 5
```
[the remark this answers is not in the document]
Comment (hanzei) [reply model/link_metadata.go]: Fixed in 12cce31b1.

File: model/license.go (hunk 9 above)
[the remark this answers is not in the document]
Comment (hanzei) [reply model/license.go]: Removed them in 92d112f38.

File: model/channel.go [hunk 11]
```diff
@@ -15,22 +15,23 @@ import (
...
-	CHANNEL_SORT_BY_USERNAME = "username"
-	CHANNEL_SORT_BY_STATUS   = "status"
+	ChannelTypeOpen    = "O"
+	ChannelTypePrivate = "P"
+	ChannelTypeDirect  = "D"
+	ChannelTypeGroup   = "G"
+
+	ChannelGroupMaxUsers       = 8
+	ChannelGroupMinUsers       = 3
+	DefaultChannel             = "town-square"
```
[the remark this answers is not in the document]
Comment (hanzei) [reply model/channel.go]: Fixed in 2146dd944.

File: model/terms_of_service.go [hunk 12]
```diff
@@ -11,7 +11,7 @@ import (
 	"unicode/utf8"
 )
 
-const TERMS_OF_SERVICE_CACHE_SIZE = 1
+const TermsOfServiceCacheSize = 1
```
[the remark this answers is not in the document]
Comment (hanzei) [reply model/terms_of_service.go]: Good catch :+1: Remove in 44b34d310.

File: model/client4.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (agnivade) [reply model/client4.go:40]: Sounds good.

**Pull request merged**